vizro-core/src/vizro/models/_page.py

```diff
@@ -136,18 +136,25 @@ def build(self):
         return html.Div([control_panel, components_container])
 
     def _update_graph_theme(self):
+        # The obvious way to do this would be to alter pio.templates.default, but this changes global state and so is
+        # not good.
+        # Putting graphs as inputs here would be a nice way to trigger the theme change automatically so that we don't
+        # need the update_layout call in Graph.__call__, but this results in an extra callback and the graph
+        # flickering.
+        # TODO: consider making this clientside callback and then possibly we can remove the update_layout in
+        #  Graph.__call__ without any flickering.
+        # TODO: consider putting this as dashboard level like with _update_theme (though possibly pointless because you
+        #  don't see graphs on other pages and then when you change page it always redraws graphs anyway).
+        # TODO: consider putting the Graph-specific logic here in the Graph model itself (whether clientside or
+        #  serverside) to keep the code here abstract.
```

Comment:
I think only the first `TODO` looks like a candidate for improvement.

For the second one, you've already described why it's not a good idea.

I can't see how to implement the last TODO you wrote without applying all the filters, parameters again (which is slow).

Reply:
On the last TODO, my thinking was that the graph-specific logic like `patched_figure["layout"]["template"] = themes.dark if theme_selector_on else themes.light` should ideally live in the `Graph` model, and we would somehow try to simplify the `ifinstance(component, Graph)`. The callback itself would still live here like it does now.

It would just be a bit of refactoring, not changing the number of callbacks or anything like that.